core/src/wallet/cosmos/explorers/StargateRpcsParsers.hpp

```diff
@@ -0,0 +1,907 @@
+        // Debug safety assertion
+        assert((firstReward.GetObject()[kDenom].GetString() == currency.units[0].code));
+
+        parseCoin(
+            firstReward,
+            out.pendingReward);  // Assuming only one reward per validator
+    }
+}
+
+/// Parse the relevant informations from /blocks/{latest || height} into cosmos::Block.
+/// The `currencyName` is necessary context for LibCore but not present in the JSON,
+/// therefore it is injected by the caller who knows the currency context
+template <class T>
+void parseBlock(const T &node, const api::Currency &currency, cosmos::Block &out)
+{
+    out.currencyName = currency.cosmosLikeNetworkParameters.value().Identifier;
+    assert((node.HasMember(kBlockId)));
+    out.hash = node[kBlockId].GetObject()[kHash].GetString();
+    assert((node.HasMember(kBlock)));
+    auto blockMetaNode = node[kBlock].GetObject();
+    assert((blockMetaNode.HasMember(kHeader)));
+    out.height =
+        BigInt::fromString(blockMetaNode[kHeader].GetObject()[kHeight].GetString())
+            .toUint64();
+    out.time =
+        DateUtils::fromJSON(blockMetaNode[kHeader].GetObject()[kTime].GetString());
+}
+
+template <typename T>
+void parseProposalContent(const T &node, cosmos::ProposalContent &out)
+{
+    assert((node.HasMember(kType)));
+    assert((node.HasMember(kDescription)));
+    assert((node.HasMember(kTitle)));
+    out.type = node[kType].GetString();
+    out.descr = node[kDescription].GetString();
+    out.title = node[kTitle].GetString();
+}
+
+template <typename T>
+void parseFee(const T &node, cosmos::Fee &out)
+{
+    assert((node.HasMember(kGas)));
+    assert((node.HasMember(kAmount)));
+    out.gas = BigInt::fromString(node[kGas].GetString());
+    if (node[kAmount].IsArray()) {
+        const auto &amountArray = node[kAmount].GetArray();
+        out.amount.assign((std::size_t)amountArray.Capacity(), cosmos::Coin());
+        auto index = 0;
+        for (const auto &aNode : amountArray) {
+            parseCoin(aNode.GetObject(), out.amount[index]);
+            index += 1;
+        }
+    }
+}
+
+template <class T>
+void parseAccount(const T &accountNode, cosmos::Account &account)
+{
+    assert((accountNode.HasMember(kAccount)));
+    const auto &node = accountNode[kAccount].GetObject();
+    assert((node.HasMember(kGrpcType)));
+    account.type = node[kGrpcType].GetString();
+
+    assert((node.HasMember(kAccountNumber)));
+    assert((node.HasMember(kSequence)));
+    assert((node.HasMember(kAddress)));
+    account.accountNumber = node[kAccountNumber].GetString();
+    account.sequence = node[kSequence].GetString();
+    account.address = node[kAddress].GetString();
+
+    assert((node.HasMember(kPubKey)));
+    // Pubkey is null for accounts that only received funds
+    // "pub_key": {
+    //   "@type": "/cosmos.crypto.secp256k1.PubKey",
+    //   "key": "AnDGSF7VFQs9VKEceS6cWhfajxh/mmv+Mbu9bCM9BDzK"
+    // },
+    if (node[kPubKey].IsObject()) {
```

Comment:
If pubkey can be null maybe the `assert()` is too much here, and the `HasMember` should be put into the `if()`?

Reply:
I meant that if the account only received funds, the json is 
```json
"pub_key": null
```
so node[kPubKey] exists, but is not an object.